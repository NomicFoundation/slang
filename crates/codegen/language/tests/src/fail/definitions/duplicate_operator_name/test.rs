#![allow(unused_crate_dependencies)]

codegen_language_macros::compile!(
    name = Foo,
    root_item = Bar1,
    leading_trivia = Sequence([]),
    trailing_trivia = Sequence([]),
    versions = ["1.0.0"],
    sections = [Section(
        title = "Section One",
        topics = [Topic(
            title = "Topic One",
            items = [
                Struct(
                    name = Bar1,
                    fields = (
                        field1 = Required(NonTerminal(Bar2)),
                        field2 = Required(NonTerminal(Bar3))
                    )
                ),
                Precedence(
                    name = Bar2,
                    operators = [
                        Operator(
                            expression_name = Op1,
                            model = Prefix,
                            fields = (operator = Required(NonTerminal(Bar2)))
                        ),
                        Operator(
                            expression_name = Op1,
                            model = Prefix,
                            fields = (operator = Required(NonTerminal(Bar2)))
                        )
                    ],
                    primary_expressions = []
                ),
                Precedence(
                    name = Bar3,
                    operators = [
                        Operator(
                            expression_name = Op1,
                            model = Prefix,
                            fields = (operator = Required(NonTerminal(Bar3)))
                        ),
                        Operator(
                            expression_name = Op2,
                            model = Prefix,
                            fields = (operator = Required(NonTerminal(Bar3)))
                        )
                    ],
                    primary_expressions = []
                )
            ]
        )]
    )]
);

fn main() {}
