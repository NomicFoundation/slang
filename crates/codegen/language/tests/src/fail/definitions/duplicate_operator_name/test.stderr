error: Expression 'Op1' is already defined under a different item.
  --> src/fail/definitions/duplicate_operator_name/test.rs:41:47
   |
41 | ...                   expression_name = Op1,
   |                                         ^^^
